src/layouts/Resources.jsx

```diff
@@ -1,378 +1,166 @@
-  constructor(props) {
-    super(props);
-    this.state = {
-      resourceCategories: [],
-      resourcePages: undefined,
-      categoryModalIsActive: false,
-      settingsIsActive: false,
-      resourceRoomName: 'resource room',
-      newResourceRoomName: '',
-      selectedResourcePage: {
-        isNewPost: true,
-        category: '',
-        fileName: '',
-      },
-      isSelectingFile: false,
-      isFileStagedForUpload: false,
-    };
-  }
-
-  async componentDidMount() {
-    try {
-      const { match } = this.props;
-      const { siteName } = match.params;
-
-      // Get the resource categories in the resource room
-      const resourcesResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources`, {
-        withCredentials: true,
-      });
-      const { resourceRoomName, resources: resourceCategories } = resourcesResp.data;
+// Import utils
+import { prettifyResourceCategory } from '../utils';
 
-      if (resourceRoomName === undefined) {
-        this.setState({ resourceRoomName });
-      } else {
-        // Obtain the title, date, fileName, category for resource pages across all categories
-        const resourcePagesArray = await Bluebird.map(resourceCategories, async (category) => {
-          const resourcePagesResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources/${category.dirName}`, {
-            withCredentials: true,
-          });
-          const { resourcePages } = resourcePagesResp.data;
+// axios settings
+axios.defaults.withCredentials = true
 
-          if (resourcePages.length > 0) {
-            return resourcePages.map((resourcePage) => {
-              const { title, date } = retrieveResourceFileMetadata(resourcePage.fileName);
-              return {
-                title,
-                date,
-                fileName: resourcePage.fileName,
-                category: category.dirName,
-              };
-            });
-          }
+// Constants
+const BACKEND_URL = process.env.REACT_APP_BACKEND_URL
 
-          return undefined;
-        }, { concurrency: 10 });
+const Resources = ({ match, location }) => {
+  const { siteName } = match.params;
 
-        const resourcePages = _.compact(_.flattenDeep(resourcePagesArray));
+  const [isLoading, setIsLoading] = useState(true)
+  const [resourceRoomName, setResourceRoomName] = useState()
+  const [newResourceRoomName, setNewResourceRoomName] = useState('')
+  const [resourceFolderNames, setResourceFolderNames] = useState()
 
-        this.setState({ resourceRoomName, resourceCategories, resourcePages });
+  useEffect(() => {
+    try {
+      const fetchData = async () => {
+        // Get the resource categories in the resource room
+        const resourcesResp = await axios.get(`${BACKEND_URL}/sites/${siteName}/resources`);
+        const { resourceRoomName, resources: resourceCategories } = resourcesResp.data;
+
+        if (resourceRoomName) {
```

Comment:
Check that `resourceCategories` has length > 0

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/207/commits/d91a4e0458b27fa5f2223db4d1df4ebcf78cccac